### PR TITLE
[146100] Remove gulp entry point

### PR DESCRIPTION
**Pull request by ttokit**
https://acsmine.tok.access-company.com/redmine/issues/146100

### CODE REVIEW

**Comment by skirino** [comment]
We have to revise the whole moduledoc as now there's only 1 method to build static assets.

**Comment by ttokit** [comment]
The moduledoc is fixed in f4c32fc.
It seems to me enough, but does we need a larger fix?

**Comment by skirino** [comment]
> It seems to me enough, but does we need a larger fix?

Yes :)
In my previous comment "there's only 1 method to build static assets", I meant that now we shouldn't say "asset preparation methods" (note the plural form).
Please update the doc accordingly.
For example, [this paragraph](https://github.com/access-company/antikythera/pull/37/files#diff-260b36156dd463c4864564187cc8f23bR18) should be revised.

Sure, there _are_ 2 methods to manage packages (yarn and npm), but they are not for asset preparations; they are for, well, package management.
With respect to the main objective of this mix task (asset preparation), the choice of "yarn or npm" isn't a big deal; in this moduledoc just saying "use yarn if yarn.lock exists, otherwise use npm" would be fine.
(And actually I'm in favor of deprecating npm for package management in all gear projects, but it's another story...)

File: lib/mix/prepare_assets.ex [hunk 1]
```diff
@@ -112,19 +90,15 @@ defmodule Mix.Tasks.Antikythera.PrepareAssets do
...
   end
 
-  defunp find_build_step_in_project(env :: v[String.t]) :: nil | :npm_script | :gulp do
-    cond do
-      npm_script_available?(env)  -> :npm_script
-      File.exists?("gulpfile.js") -> :gulp
-      true                        -> nil
-    end
+  defunp find_build_step_in_project(env :: v[String.t]) :: nil | :npm_script do
+    if npm_script_available?(env), do: :npm_script, else: nil
```
Comment (skirino) [inline lib/mix/prepare_assets.ex]: Now we don't have to convert the boolean value returned by `npm_script_available?(env)` into `nil | :npm_script`.
Remove `find_build_step_in_project/1` and simply use `if-else` in `run_impl/1`.

**Comment by skirino** [comment]
In `install_packages!/1`, we've already confirmed that `package.json` file is present.
Simply checking existence of `yarn.lock` file should be enough here.

```ex
if File.exists?("yarn.lock") do
  run_command!("yarn", [], env)
else
  remove_node_modules_if_dependencies_changed!()
  run_command!("npm", ["install"], env)
end
```

**Comment by skirino** [comment]
Not directly related to this PR's changeset, but in `npm_script_available?/1` is it OK to depend on the output format of `npm run`?
I think it's more robust to JSON-parse content of `package.json` and check `scripts` => `antikythera_prepare_assets`.

File: lib/mix/prepare_assets.ex (hunk 1 above)
Comment (ttokit) [reply lib/mix/prepare_assets.ex]: Fixed in 68a874a

**Comment by ttokit** [comment]
`install_packages!/1` is fixed in 04b711b

**Comment by ttokit** [comment]
`npm_script_available?/1` is fixed in #38

**Comment by skirino** [comment]
Sorry for my late feedbacks.
Please address the newly added points and merge when you finish.

File: lib/mix/prepare_assets.ex [hunk 2]
```diff
@@ -26,36 +24,26 @@ defmodule Mix.Tasks.Antikythera.PrepareAssets do
   (see also `Antikythera.Env`).
```
[the remark this answers is not in the document]
Comment (ttokit) [reply lib/mix/prepare_assets.ex]: Fixed in 5799e98

File: lib/mix/prepare_assets.ex [hunk 3]
```diff
@@ -13,11 +13,9 @@ defmodule Mix.Tasks.Antikythera.PrepareAssets do
   See `Antikythera.Asset` for details.
 
   If your gear uses some kind of preprocessing tools to generate asset files (JS, CSS, etc.),
-  you have to set up any of the supported asset preparation methods described in the next section.
+  you have to set up the supported asset preparation method described in the next section.
 
-  This mix task detects available asset preparation methods in your gear repository, then executes them in sequence.
-  If any part of the preparation resulted in failure (non-zero exit code),
-  the whole task will abort and thus auto-deploy will fail.
+  This mix task checks if asset preparation method in your gear repository is available, then execute it if possible.
```
[the remark this answers is not in the document]
Comment (ttokit) [reply lib/mix/prepare_assets.ex]: Fixed in 5799e98

File: lib/mix/prepare_assets.ex [hunk 4]
```diff
@@ -26,36 +24,26 @@ defmodule Mix.Tasks.Antikythera.PrepareAssets do
...
-  #### 2. Using `npm install`
-
-  - Prerequisite: `package.json` file
-  - Command: `npm install`
+  Use [`yarn`](https://yarnpkg.com/en/) if `yarn.lock` file exists, otherwise use `npm install`.
 
   ### Build
 
-  #### 1. Using [npm-scripts](https://docs.npmjs.com/misc/scripts)
+  #### Using [npm-scripts](https://docs.npmjs.com/misc/scripts)
```
[the remark this answers is not in the document]
Comment (ttokit) [reply lib/mix/prepare_assets.ex]: Fixed in 5799e98

**Pull request merged**